### PR TITLE
doc: report which warnings/errors are new

### PR DESCRIPTION
**Pull request by nashif**
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Comment by dbkinder** [comment]
The problem is that the API doc build (sphinx/breathe) has a known issue parsing nested unnamed structs/unions and generates a warning.  @inakypg  submitted an issue against sphinx/breathe (https://github.com/sphinx-doc/sphinx/issues/2683) over a year ago and the issue remains today.  If an end user (or another developer) generates documentation, they will get these warnings.

Until we can generate a clean doc build with no warnings, having the build script write out:

    New errors/warnings found, please fix them

showing these "expected" warnings, isn't a friendly message when these warnings aren't new and they can't fix them.  That's why they're filtered out as "expected' warnings.

There is a workaround for the sphinx/breathe issue (giving the nested unnamed struct/union a name) as Inaky suggests.  If we can actually get rid of these "expected" warnings by fixing the code , then I'd be OK with this PR.

We're seeing warnings from the definitions of ieee802154_req_params, net_mgmt_event_callback, net_buf, uart_device_config, bt_gatt_read_params, bt_gatt_discover_params, bt_conn_info, and in6_addr.  Can those get "fixed" with the proposed workaround?

Otherwise, I think PR #4890 is the way to go.  It has a fix (prefixing a ``-`` to the sphinx-build line) that prevented the filtering script from running because the sphinx-build returned an error that stopped make.

**Comment by dbkinder** [comment]
Sigh.  I should have looked at the code a bit closer.  You're only writing out the filtered (unexpected) errors here at the end.  I still don't like the ``tee`` writing out all the expected warnings, but this indeed might be a compromise.  :)

File: doc/scripts/filter-doc-log.sh [hunk 1]
```diff
@@ -11,15 +11,25 @@ CONFIG_DIR=${ZEPHYR_BASE}/.known-issues/doc
...
-      $KI_SCRIPT --config-dir ${CONFIG_DIR} ${LOG_FILE}
+if [ -s "${LOG_FILE}" ]; then
+   $KI_SCRIPT --config-dir ${CONFIG_DIR} ${LOG_FILE} > doc.warnings 2>&1
+   if [ -s doc.warnings ]; then
+	   echo
+	   echo -e "${red}New errors/warnings found, please fix them:"
+	   echo -e "=============================================="
+	   tput sgr0
+	   echo
+	   cat doc.warnings
```
Comment (dbkinder) [inline doc/scripts/filter-doc-log.sh]: Should add ``doc.warnings``to .gitignore then.

File: doc/scripts/filter-doc-log.sh [hunk 2]
```diff
@@ -11,15 +11,25 @@ CONFIG_DIR=${ZEPHYR_BASE}/.known-issues/doc
...
+	   echo
```
Comment (dbkinder) [inline doc/scripts/filter-doc-log.sh]: How about adding an else after this indicating there were no "new" errors/warnings:

**Review by dbkinder — APPROVED** [review]
I'm OK with this compromise :)

File: doc/scripts/filter-doc-log.sh (hunk 2 above)
Comment (nashif) [reply doc/scripts/filter-doc-log.sh]: yes, good idea

**Pull request merged**